2-mandatory/1-writers.js

```diff
@@ -59,60 +59,76 @@ Exercise 1:
   and insert the corresponding values to the place holders that are indicated in curly braces:
   "Hi, my name is {firstName} {lastName}. I am {age} years old, and work as a {occupation}."
 */
-function logAllWriters() {
-  // write your code to log all writers here
+function logAllWriters(obj) {
+   let newObj=  obj.map(x => {
+  return  console.log(
+    `Hi, my name is ${x.firstName} ${x.lastName}. I am ${x.age} years old, and work as a ${x.occupation}.`
+  );
+  
+});
+return; newObj 
 };
-
+logAllWriters(writers);
 /*
 Exercise 2:
   Only `console.log()` out the writers who are in their 40s (meaning between 40 and 49)
   and not alive anymore. Use the below sentence format:
   "Writer {firstName} {lastName} died at {age} years old."
 */
 
-function logDeadWritersInTheirForties() {
-  // write your code here
+function logDeadWritersInTheirForties(ages) {
+let diedAt=ages.filter(x=>{
+ if( x.age >= 40 && x.age <= 49 && x.alive ==false ){
+console.log(`Writer ${x.firstName} ${x.lastName} died at ${x.age} years old.`);
+ } 
+})
 }
-
+logDeadWritersInTheirForties(writers);
 /*
 Exercise 3:
   Only `console.log()` out alive writers who are in their 40s (meaning between 40 and 49):
   "Hi, my name is {firstName} {lastName}. I am {age} years old."
 */
 
-function logAliveWritersInTheirForties() {
-  // write your code here
+function logAliveWritersInTheirForties(obj) {
+  let aliveWiters=obj.filter(x=>{
+     if( x.age >= 40 && x.age <= 49 && x.alive== true ){
+      console.log(
+        `Hi, my name is ${x.firstName} ${x.lastName}. I am ${x.age} years old.`
```

Comment:
Again your function is not returning anything and you only logging the result using `console.log`